plaid/src/apis/npm/mod.rs

```diff
@@ -0,0 +1,117 @@
+mod hashes;
+pub mod npm_cli_client;
+pub mod npm_web_client;
+
+use std::sync::Arc;
+
+use regex::Regex;
+use reqwest::{cookie::Jar, Client};
+use serde::{Deserialize, Serialize};
+
+/// Client for interacting with npm that groups a client for CLI operations
+/// and one for web operations
+pub struct Npm {
+    config: NpmConfig,
+    client: Client,
+    cookie_jar: Arc<Jar>,
+}
+
+impl Npm {
+    pub fn new(config: NpmConfig) -> Self {
+        let cookie_jar = Arc::new(Jar::default());
+        let client = Client::builder()
+            .cookie_provider(cookie_jar.clone())
+            .build()
+            .map_err(|_| NpmError::GenericError).unwrap(); // TODO @obelisk OK to unwrap here? If we cannot build the client, perhaps we should just panic?
+        Self {
+            config,
+            client,
+            cookie_jar,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+/// Credentials and secrets for interacting with npm
+pub struct NpmConfig {
+    /// Username for the npm account
+    pub username: String,
+    /// Password for the npm account
+    pub password: String,
+    /// Secret for the TOTP-based 2FA on the npm account
+    pub otp_secret: String,
```

Comment:
This should probably be optional because not all orgs enforce this

Reply:
Good point, done